pkg/plugins/manager/loader/loader.go

```diff
@@ -195,6 +198,14 @@ func (l *Loader) loadPlugins(ctx context.Context, class plugins.Class, pluginJSO
 			return nil, err
 		}
 		metrics.SetPluginBuildInformation(p.ID, string(p.Type), p.Info.Version, string(p.Signature))
+
+		if errDeclareRoles := l.roleRegistry.DeclarePluginRoles(ctx, p.ID, p.Name, p.Roles); errDeclareRoles != nil {
+			l.log.Warn("Declare plugin roles failed",
+				"pluginID", p.ID,
+				"warning", "Make sure the role declaration is correct.",
```

Comment:
I would suggest appending this to the log message rather than as a separate key `warning`. In addition, if this is a message to operators (the ones hosting/running Grafana) they cannot really change the role declaration - the plugin/plugin developer needs to change it :) So maybe can change the message somehow? 

Moving forward, the plugin submission review and tooling (for internal plugin submission) could possibly validate the role declaration is correct?

Reply:
> Moving forward, the plugin submission review and tooling (for internal plugin submission) could possibly validate the role declaration is correct?

Yes it could and I think it would be best. Where is this tooling? Maybe we can tackle that in another issue?